container.go

```diff
@@ -30,6 +31,7 @@ type Container interface {
 	Endpoint(context.Context, string) (string, error)               // get proto://ip:port string for the first exposed port
 	PortEndpoint(context.Context, nat.Port, string) (string, error) // get proto://ip:port string for the given exposed port
 	Host(context.Context) (string, error)                           // get host where the container port is exposed
+	InspectContainer(context.Context) (*types.ContainerJSON, error) // get the JSON representing the container
```

Comment:
This can not be merged. At the moment I would like to keep the Container interface provider agnostic. Because I would like to keep the door open to eventually have a containerd provider.

exposing the inspect response will prevent this to happen.
The rest of the PR is fine, if you can remove this I will merge.

Reply:
Sure, I'll rework this to remove the inspect part

Thanks!!

Reply:
Hey @gianarb I removed the expose part, keeping the internal cache removal.

Hope it's good to go!